File: src/Web/ClientApp/src/app/planning/model/plan-entry.model.ts
import { PlanType } from './plan-type.enum';
import { PlanRepeatOn } from './plan-repeat-on.enum';
import { PersistentState } from './persistent-state.enum';

export interface PlanEntryModel {
  id?: number | undefined;
  eventDate: string;
  planType: PlanType;
  amount: number;
  description: string;
  repeatOn: PlanRepeatOn;
  endDate: string | undefined | null;
  persistentState: PersistentState;
}

File: src/Application.Abstractions/Planning/PlanEntryModel.cs
using TRS.FinalPlantasy.Application.Abstractions.Repositories;
using TRS.FinalPlantasy.Domain.Model.Planning;

namespace TRS.FinalPlantasy.Application.Abstractions.Planning;

public class PlanEntryModel : PersistedModel
{
    public int? Id { get; set; }

    public PlanTypeId? PlanType { get; set; }

    public DateOnly? EventDate { get; set; }

    public double? Amount { get; set; }

    public PlanRepeatOnId? RepeatOn { get; set; }

    public string? Description { get; set; }

    public DateOnly? EndDate { get; set; }
}
